### PR TITLE
Run builds in travis on xenial

### PR DESCRIPTION
**Pull request by webframp**
Switching to xenial since it's a newer base and has support for the
addons needed to run chef integration testing.

bionic doesn't yet appear to be added to the approved list of resources
for the apt addon:

* https://github.com/travis-ci/apt-source-safelist/tree/master/keys

Additionally switched package from chefdk to chef-workstation to stay up
to date with what's maintained. Same tools, different name.

--

#### Purpose

Update the base platform used for travis builds.

#### Known Compatibility Issues

None

### CODE REVIEW

**Comment by webframp** [comment]
Saw failures like this: https://travis-ci.org/sensu/sensu-go-chef/jobs/624832661

Small attempt to fix these

**Comment by webframp** [comment]
@majormoses Don't know why appveyor started failing here

**Comment by majormoses** [comment]
is `rspec` not in chef-workstation? https://travis-ci.org/sensu/sensu-go-chef/jobs/624854826#L257

**Comment by webframp** [comment]
> is `rspec` not in chef-workstation? https://travis-ci.org/sensu/sensu-go-chef/jobs/624854826#L257

My vault, just used wrong bin path

**Comment by webframp** [comment]
@cwjohnston or @derekgroh maybe can comment on windows failures here. I don't think I can do much about those.

File: .travis.yml [hunk 1]
```diff
@@ -1,42 +1,43 @@
...
 
-# install the pre-release chef-dk.  Use chef-stable-precise to install the stable release
+# Install the pre-release chef tools
 addons:
   apt:
     sources:
-      - chef-current-trusty
+      - chef-current-xenial
     packages:
-      - chefdk
```
Comment (derekgroh) [inline .travis.yml]: Is this value correct? 
https://github.com/travis-ci/travis-cookbooks/blob/master/.travis.yml

**Comment by derekgroh** [comment]
Thinking a bit more on #59 and this change, could we move testing to Travis.ci for both platforms? Anyone have experience with breaking down the logic for Travis.ci now that Windows support is more supported?

**Comment by webframp** [comment]
@derekgroh I'm open to that option and doing some of that work if it's agreeable. 

Think it's reasonable we do that instead of merging this as is and just ignoring appveyor failures?

**Pull request merged**